Pipeline job with extract_vba_source.yaml fails when it is triggered by a Pull Request.

Azure Pipeline using the YAML file https://github.com/takeruko/extract_vba_source/blob/e1fa3d1e366130d8a69f056e889d66da9d451f3f/yaml/azure-pipelines-extract-vba.yml
fails with the following error message.

```powershell
Starting: Set the configuration for git commit/push. 
============================================================================== 
Task : Command line 
Description : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows 
Version : 2.182.0 
Author : Microsoft Corporation 
Help : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line 
============================================================================== 
Generating script. 
========================== Starting Command Output =========================== 
/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/f70xxxxxxxxxxxxxxxxxxxxxxx.sh 
Set Requestor as XXXXX XXXXX xxxxxx@foo.com 
Checkout for branch refs/pull/256/merge 
Checkout for branch-name refs/pull/256/merge 
Fetching origin 
From https://dev.azure.com/xxxxx/XXXXXXXXXXXXX/_git/XXXXXXXXXXXXX 
 * branch refs/pull/256/merge -> FETCH_HEAD 
error: pathspec 'refs/pull/256/merge' did not match any file(s) known to git 
##[error]Bash exited with code '1'. 
Finishing: Set the configuration for git commit/push.
```

This issue is caused due to a change in the specification of pre-defined variables in Azure Pipelines.
Now, when the pipeline run by a Pull Request, the pre-defined variable `Build.SourceBranch` is set to a value based on its Pull Request ID such as `refs/pull/1/merge`.
https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services

For an alternative, it should use `System.PullRequest.SourceBranch` to get a source branch name of a Pull Request.

--- a/yaml/azure-pipelines-extract-vba.yml
+++ b/yaml/azure-pipelines-extract-vba.yml
@@ -26,13 +26,21 @@ steps:
   clean: true
 
 - script: |
-    echo Set Requester as $(Build.RequestedFor) $(Build.RequestedForEmail)
+    echo Set Requestor as $(Build.RequestedFor) $(Build.RequestedForEmail)
     git config --global user.email $(Build.RequestedForEmail)
     git config --global user.name $(Build.RequestedFor)
-    export branchname=$(echo $(Build.SourceBranch) | sed s@refs/heads/@@)
+    if ["$(System.PullRequest.SourceBranch)" = ""]; then
+      echo Checkout for branch $(Build.SourceBranch)
+      export branchname=$(echo $(Build.SourceBranch) | sed s@refs/heads/@@)
+    else
+      echo Checkout for PR source branch $(System.PullRequest.SourceBranch)
+      export branchname=$(echo $(System.PullRequest.SourceBranch) | sed s@refs/heads/@@)      
+    fi
     echo Checkout for branch-name $branchname
+    git remote update
+    git fetch origin $branchname
     git checkout $branchname
-  displayName: 'Configuration to git commit/push.'
+  displayName: 'Set the configuration for git commit/push.'
 
 - script: |
     echo Install pipenv

